doctr/utils/lang_detect.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2022, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+from typing import Tuple
+
+from lingua import Language, LanguageDetectorBuilder
+
+from doctr.datasets.vocabs import LANGUAGES
+
+__all__ = ['get_language']
+
+languages = [Language._member_map_[item.upper()] for item in LANGUAGES if
+             item.upper() in Language._member_map_.keys()]
+
+
+def get_language(text: str) -> Tuple[str, float]:
+    """Get languages of a text using lingua library.
+    Get the language with the highest probability or no language if only a few words or no language was identified
+    Args:
+        text (str): text
+    Returns:
+        The detected language in ISO 639 code and confidence score
+    """
+
+    model = LanguageDetectorBuilder.from_languages(*languages).build()
```

Comment:
@aminemindee i think we should init the model outside of the function otherwise it will be initialized again with every call

Reply:
i think this takes also some time maybe an overall speed up ?

Reply:
Done and pushed. 
we gain 0.3 s :+1: 

model | timeit value | start - end time
-- | -- | --
None | 12.6 s ± 25.1 ms | 12.55 s
fasttext | 12.5 s ± 149 ms | 12.79 s
langdetect | 12.3 s ± 96.5 ms | 12.57 s
lingua-py | 13.1 s ± 100 ms | 13.28 s
lingua-py selected languages | 12 s ± 86.1 ms | 12.22

Reply:
can you tell me where is the `with_low_accuracy_mode` i don't seem to find it.

Reply:
https://github.com/pemistahl/lingua-py#95-low-accuracy-mode-versus-high-accuracy-mode

and

https://github.com/pemistahl/lingua-py/blob/305d001a331c01caa386e7a259c7ee9f98f99d4b/lingua/builder.py#L193

Reply:
puh .. than we should test that `with_low_accuracy_mode` is enough

Reply:
with `with_low_accuracy_mode` the sentence 'this is a test sentence.' get predicted as portuguese x) the RAM goes down to 0.3 gb. 

lang_detect use also 0.3 gb of RAM and it uses an AI based approach meaning the confidence score is more accurate as a probability and not a distance.

Reply:
Mh i would say that 0.8 gb is to much .. so in fact of your tests do you think it is ok switching to lang_detect ? Let's try this and than let me know your decision i will go with it :)

Reply:
From what i tested: 
- fasttext is the best one performance wise but you have to download model and it's the larger one. 
- langdetect is the easiest one to use but it's not really well done ( you have to do a try except because he return error if the text doesn't have any character he knows) 
- lingua seems really good but uses too much RAM and it's performance is not that good with the low accuracy mode. 

So for now i think langdetect is the reasonable choice, knowing we can change it easily after.